src/back/wgsl/writer.rs

```diff
@@ -53,13 +52,25 @@ struct FunctionCtx<'a> {
     info: &'a FunctionInfo,
     /// The expression arena of the current function being written
     expressions: &'a Arena<Expression>,
+    /// Map of expressions that have associated variable names
+    named_expressions: &'a FastHashMap<Handle<Expression>, String>,
```

Comment:
looks like this type could be a typedef somewhere in lib.rs? it's already used a bunch